### PR TITLE
[crossplane] Initial integration

### PR DESCRIPTION
**Pull request by AdamKorcz**
Crossplane is a framework for building cloud native control planes without needing to write code. It has a highly extensible backend that enables users to build a control plane that can orchestrate applications and infrastructure no matter where they run, and a highly configurable frontend that puts you in control of the schema of the declarative API it offers.

Crossplane is a [CNCF](https://www.cncf.io/) project. It is used by companies like:

- Alibaba
- Redhat
- GitLab
- Microsoft
[[reference](https://crossplane.io/#:~:text=Started%20by%20Upbound%20and%20adopted%20by%20the%20Cloud%20Native%20community)]

____________________________________________________________

@hasheddan @negz @muvaf Are you interested in integrating Crossplane into OSS-fuzz for continuous fuzzing?
This PR adds the build files and the first fuzzer to get fuzzing started.

### CODE REVIEW

**Comment by negz** [comment]
@AdamKorcz I'm not familiar with oss-fuzz but this sounds like a good idea. I'll read up on oss-fuzz now.

File: projects/crossplane/fuzz.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	"github.com/crossplane/crossplane-runtime/pkg/parser"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func FuzzParse(data []byte) int {
+	objScheme := runtime.NewScheme()
+	metaScheme := runtime.NewScheme()
+	p := parser.New(metaScheme, objScheme)
+	r := ioutil.NopCloser(bytes.NewReader(data))
+	_, _ = p.Parse(context.TODO(), r)
```
Comment (muvaf) [inline projects/crossplane/fuzz.go:32]: I'm guessing what's tested here is whether the program crashes with arbitrary input, right?

**Review by muvaf — COMMENTED** [review]
@negz Quoting from linked [wiki](https://en.wikipedia.org/wiki/Fuzzing):

> Fuzzing or fuzz testing is an automated software testing technique that involves providing invalid, unexpected, or random data as inputs to a computer program. The program is then monitored for exceptions such as crashes, failing built-in code assertions, or potential memory leaks.

Seems like this PR sends arbitrary data to our package parser to see if it crashes or leaks memory. LGTM.

File: projects/crossplane/fuzz.go (hunk 1 above)
Comment (AdamKorcz) [reply projects/crossplane/fuzz.go:32]: That is correct.

**Comment by negz** [comment]
> @negz Quoting from linked wiki:

Haha, right. I'm familiar with fuzzing as a practice - just not this specific project. :)

**Comment by muvaf** [comment]
@negz Ah I wasn't, that's probably why I thought you were asking that :)

**Comment by AdamKorcz** [comment]
@muvaf @negz Fuzzing was discussed on the last meeting. I am tagging you to hear if we can complete this PR to get fuzzing started for Crossplane. All we need is (at least) one maintainers email address to add to the `project.yaml` file.

**Comment by negz** [comment]
@AdamKorcz Please do add me! I believe you can get my email address from my GitHub profile. :)

**Comment by muvaf** [comment]
@AdamKorcz You can add me as well, me@muvaf.com

File: projects/crossplane/project.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+homepage: "https://github.com/crossplane/crossplane"
+main_repo: "https://github.com/crossplane/crossplane"
+primary_contact: ""
```
Comment (jbw976) [inline projects/crossplane/project.yaml:3]: looks like @muvaf and @negz have given the go ahead to use their email address as maintainers for crossplane in this file - this initial config seems good to me after that!

**Review by jbw976 — APPROVED** [review]
Thanks @AdamKorcz, we're looking forward to having the fuzz testing going soon!

**Review by DavidKorczynski — APPROVED** [review]

**Review by jonathanmetzman — APPROVED** [review]
lgtm

**Pull request merged**